### ISSUE TITLE
Support enumeratedValues in registers

### ISSUE THREAD

**Issue by chrisnappi** [Origen-SDK/o2#31]
IPXact supports bitfield enumerated values:
https://accellera.org/images/xmlschema/spirit/1685-2009/memoryMap.xsd
search for:
<xs:element name="enumeratedValues">

We should support this, so that we can do things like:
dut.reg(:clock_ctrl).bits(:clock_select).write(:osc_24mhz)

**Comment by ginty** [comment]
Yes this will definitely be done, the data structure for it is already there - https://github.com/Origen-SDK/o2/blob/master/rust/origen/src/core/model/registers.rs#L124

Just need to come up with the appropriate API.
